### DISCUSSION TITLE
Help locating Truenas code - Truenas Core 13.0_U6.2

### DISCUSSION THREAD

**Discussion by wmsunder** [community/community#142824] [Programming Help]
### Select Topic Area

Question

### Body

Hello friends. I stupidly deleted:

client.py
from /usr/local/lib/python3.9/site-packages/middlewared/client/  

schema.py
from /usr/local/lib/python3.9/site-packages/middlewared/

cloud_sync.py
from /usr/local/lib/python3.9/site-packages/middlewared/plugins/

I need your help locating the correct code for these files so that I may re-create the files and place the correct code in each file. Can someone please assist? Thank you - Your local village idiot...

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @wmsunder ! 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.
